lib/src/cpp/utils/XMLManager.hpp

```diff
@@ -44,34 +44,80 @@ namespace utils {
  */
 class XMLManager
 {
+    //======================//
+    // Singleton management //
+    //======================//
+
 public:
 
+    /**
+     * @brief Get the instance object of the singleton class
+     *
+     * @return XMLManager&
+     */
+    static XMLManager& get_instance()
+    {
+        static XMLManager instance;
+        return instance;
+    }
+
     /**
      * @brief Construct a new Parse XML object, which initializes Xerces required tools,
      *  and reads given xml_file document.
      *
      * @param[in] xml_file string with the file path
-     * @param[in] create_file bool (optional) create file if the flag is set
      *
-     * @throw Error exception if Xerces XML workspace could not be initialized
+     * @throw Error exception if Xerces XML workspace could not be initialized, or it was already initialized
      */
-    XMLManager(
-            const std::string& xml_file,
-            bool create_file = false);
+    void initialize(
+            const std::string& xml_file);
+
+    /**
+     * @brief Terminate XML workspace. If required to save or update the XML configuration in the filesystem,
+     *  this method must be called to save them.
+     *  If error occurred or resultant XML configuration is not valid, it will not save the XML configuration.
+     *
+     * @throw Error Exception if the XML file could not be written in filesystem
+     */
+    void terminate();
 
     /**
-     * @brief Pase XML Destructor
+     * @brief Method that checks if the XML workspace has been initialized.
      *
+     * @throw Error exception if XML workspace was not initialized
+     *
+     * @return true if XML workspace already initialized
+     * @return false if XML workspace has not been initialized
```

Comment:
I suppose this line can be removed as the exception is thrown in that specific case.